tests/unittests/core/worker/test_experiment.py

```diff
@@ -611,21 +609,160 @@ def test_configurable_broken_property(space: Space):
+        assert stats.eta is None
+        assert stats.eta_milliseconds is None
+
+
+def test_experiment_stats_max_trials_lesser_than_nb_completed(space: Space):
+    """Check stats with max trials < number of completed trials."""
+    NUM_COMPLETED = 3
+    statuses = (["completed"] * NUM_COMPLETED) + (["reserved"] * 2)
+    with OrionState(trials=generate_trials(statuses)) as cfg:
+        exp = Experiment("supernaekei", mode="x", space=space, storage=cfg.storage)
+        exp._id = cfg.trials[0]["experiment"]
+        exp.metadata = {"datetime": datetime.datetime.utcnow()}
+        exp.max_trials = 2
+        stats = exp.stats
+        assert stats.trials_completed == NUM_COMPLETED
+        assert stats.best_trials_id == cfg.trials[2]["id"]
+        assert stats.best_evaluation == 0
+        assert stats.start_time == exp.metadata["datetime"]
+        assert stats.finish_time == cfg.trials[3]["end_time"]
+        assert stats.duration == datetime.timedelta(seconds=3)
+        assert stats.whole_clock_time == datetime.timedelta(seconds=3)
+        assert stats.nb_trials == NUM_COMPLETED + 2
+        assert stats.trial_status_count == {"completed": NUM_COMPLETED, "reserved": 2}
+        # If max trials < completed trials, then ETA is 0, and progress is relative to nb trials
+        assert stats.max_trials == 2
+        assert stats.progress == 0.6
+        assert stats.eta == datetime.timedelta()
+        assert stats.eta_milliseconds == 0
+
+
+def test_experiment_stats_no_completed_trials(space: Space):
+    """Check stats with no completed trials."""
+    NUM_COMPLETED = 0
+    statuses = (["completed"] * NUM_COMPLETED) + (["reserved"] * 2)
+    with OrionState(trials=generate_trials(statuses)) as cfg:
+        exp = Experiment("supernaekei", mode="x", space=space, storage=cfg.storage)
+        exp._id = cfg.trials[0]["experiment"]
+        exp.metadata = {"datetime": datetime.datetime.utcnow()}
+        exp.max_trials = 2
+        stats = exp.stats
+        assert stats.trials_completed == NUM_COMPLETED
+        # Stats relative to completed trials cannot be computed
+        assert stats.best_trials_id is None
+        assert stats.best_evaluation is None
+        assert stats.start_time == exp.metadata["datetime"]
+        assert stats.finish_time == exp.metadata["datetime"]
+        assert stats.duration == datetime.timedelta()
+        assert stats.whole_clock_time == datetime.timedelta()
+        assert stats.nb_trials == NUM_COMPLETED + 2
+        assert stats.trial_status_count == {"reserved": 2}
+        # If no completed trials, then ETA is infinite, and progress is 0
+        assert stats.max_trials == 2
+        assert stats.progress == 0
+        assert stats.eta == "infinite"
+        assert stats.eta_milliseconds is None
+
+
+def test_experiment_stats_normal(space: Space):
+    """Check stats with max trials > completed trials."""
     NUM_COMPLETED = 3
     statuses = (["completed"] * NUM_COMPLETED) + (["reserved"] * 2)
```

Comment:
```suggestion
    NUM_COMPLETED = 3
    NUM_RESERVED = 2
    statuses = (["completed"] * NUM_COMPLETED) + (["reserved"] * NUM_RESERVED)
```

Reply:
Same thing for tests above, setting number of reserved trials using a variable.

Reply:
Ok !